### DISCUSSION TITLE
Labs is out for HERO13

### DISCUSSION THREAD

**Discussion by dnewman-gpsw** [gopro/labs#1062] [Announcements]
There is a lot in the first HERO13 Labs release, one of the bigger Labs updates in a while.

This will take you a long time to try all of this, and will likely be a few Labs bugs. So feedback and new ideas always welcome. 

Release Notes for 1.10.70 - released September 10, 2024
- Added STOP=x for Stop Motion where is x the number for frame to store per shutter press. Works great with the remote. Mode button ends capture.
- Added ONIO=x Onion percentage used for Stop Motion where is x the precentage of transparency of the onion skin.
- Added DIST=x Time-Lapse Video Distance, x is a GPS travel distance, rather than time for an in-camera computed timelapse. When x=1 10m will be used.
- Added DIVE=1 - optimizing the Hypersmooth stabilization for underwater. The water refractive index defaults to 1.335 (good for fresh and command salt water), if you know your area RI is significantly different DIVE=r where r can be 1.3 to 1.45 is supported.
- Added WARM=x and COOL=x to offset white balance. WARM is good to use with DIVE, so you can improve the red respond for deeper dives.
- Added RLTC=1 to enable LTC timecode reading from line-in audio
- Added OLTC=x frame offset to LTC, enabling calibration you are seeing any offset.
- Added MMSK=time, for auto generating a mask for the new motion detector
- Added GOAL=x - 0 to 60 (or x) MPH speed timing. Not for precision timing, just for fun. Drive safe.
- Added GPSL=x to enabled continous GPS location logging, even when not capturing, so the capture can log more of you path and adventures. Path is saved by day into MISC/GPS-yy-mm-dd.gpmf file. GPSL=1 logs everything, GPSL=x will only log movement great than x meter from the last stored location.
- Added custom Labs setting within Preset names. e.g. *DIVE=1*BITR=160 (standard formatting) or DIVE 24HZ or 24HZ,DIVE,NR01 (space or comma delimited). For advanced Labs users, preset names can call macros. Warning: don’t put permanent settings or actions with presets, could brick your camera.
- Added !A for appending frames for a new controlled video timelapse. Stop motion, GPS interval Lapse etc.
- Added !C to Clear last GSP distance so command like this are simplier >c10!C+!A!R //(if distance > 10m) Clear()+Append() repeat;
- Added support for !R-x relative gotos. e.g. “Hello”!N!R-2 keep back 2 character before !R.
- Added command p.3 for 3s interval support
- Added command p4 for a custom 4fps Time-lapse Video interval (0.25s)
- Added 24.0 LTC Timecode support
- Added !SAVEfile=0 - to erase MISC/file
- Added oNx commands for Noise Reduction controls - oN0 - High/default, oN1 - Medium, oN2 - Low.
- Added script variables b:T - battery temperature, p:T - processor temperature
- Added mVB - mode Video Burst command
- Added oX0, oX2 to oX9 - Standard, MLM2.0, MLM2.5, Macro, future, ND4, ND8, ND16, ND32
- Added r4S - new Square 4K mode for the Ulta Wide lens mode (and MLM2.)
- Added variables a:X, a:Y, a:Z and g:X, g:Y, g:Z, for accelerometer and gyro single axis measurements
- Added commands r1V and r4V, for 1080p and 4K vertical video modes.
- Improved the new style motion detection, faster and more precise. System variable m is the percentage of the frame in motion.
- Improved QR Scan will a Labs script is already running.
- Improved QR Sanning of complex codes in different lens modes.
- Improved handling of utf-8 characters in QR Codes.
- Improved long form wake reliability
- Improved archive messages
- Improved AUDS, auto start voice control (required for SPL outside of video recording)
- Fixed subroutine handling of !R repeats
- Fixed commands running too fast, occasionally missing steps or sleeps
- Fixed BOOT commands to only happen during boot, so that when scan in a new boot command, it will not run immediately
- Fixed attempted file I/O with SD Card removed.

**Comment by kouyoumin** [comment]
Good job.
GOAL seems interesting. Does it display the result on screen or log to a file?

**Reply by dnewman-gpsw** [reply]
GOAL briefly displays a new fast time on the LCD, but saves a JPEG on the result and acceleration curve to the MISC folder on the SD card.  It is a gimmicky feature, just a demo for things you can do with metadata and GPS.

**Comment by mactac** [comment]
Very cool. I think the timecode variables are really useful. I presume that it would be possible to bring down the bitrate if camera temp gets too high? That could solve some issues.
One question: Even though GoPro does not sell an ND64 ND filter, is there also a command (and corresponding magnet configuration) to set the ND filter to ND64 (or even 128?) . I use an ND64 a lot, and I realize that there may be some protectionism with the GoPro filters, but leaving out the ability to do things automatically with an ND64 will be a bit of a pain.

**Reply by dnewman-gpsw** [reply]
You could try that, temperature controlled bitrate, I have not.

There is not ND64 configuration. Just set ND32 configure for your ND64 are it will work fine.

**Reply by mactac** [reply]
> There is not ND64 configuration. Just set ND32 configure for your ND64 are it will work fine.

OK, thanks.  Will that not make the settings be a bit off if the camera thinks it's an ND32?

**Reply by dnewman-gpsw** [reply]
It will not, exposure will be fine for ND64.  It will still suggest a lower ND is the images is too dark.

**Reply by mactac** [reply]
OK, thank you.  I assume all the shutter speed/motion blur compensation stuff will work properly too?

**Reply by dnewman-gpsw** [reply]
Yes.

**Comment by kouyoumin** [comment]
> Added custom Labs setting within Preset names.

Does this mean different labs settings can be saved to presets and automatically loaded among preset change?
Say, add timestamp overlay when switching to "dashcam" preset and remove overlay when switching to "standard" preset?

**Reply by dnewman-gpsw** [reply]
Yes it is possible. It is very experimental, so expect bugs.

**Comment by Espusen** [comment]
So this should be more reliable (again) for Long form T-L??

**Reply by dnewman-gpsw** [reply]
Yes, I think so, although I haven't started any HERO13 reliability testing yet.

**Comment by skunterdownunder** [comment]
Hey all. Trying to get the latest QR code for preset Dive Mode. Does anyone have it ?

**Reply by dnewman-gpsw** [reply]
![image](https://github.com/user-attachments/assets/2a30a82b-89e8-4db8-891b-21fb9c71ad95)

This creates the preset called "DIVE", the only keyword that does it.  This preset should be Wide, and HS on, otherwise you can set anything else, e.g. resolution, framerate, Protune settings are all your choice.

**Reply by skunterdownunder** [reply]
Legend. Thanks mate

**Comment by bergie5737** [comment]
Hi. I'm lost with the DIST 4CC code. I managed to get it working. But it appears that it sets all time lapse presets to distance? I created a preset called "Distance" Am I doing something wrong? Here is the QR I used to scan. mTp1oMPRES="1,Distance"mTr4XtcNd0b1i4M1S180g1oMDIST=1

My use case is,  I want to record us skiing. Timelapse could be too slow, because us beginners spent more time waiting than skiing! At this stage it appears I have to rescan the QR code every time I want to record a distance timelapse. Storing it permanently messed things up so bad, I had to reset all presets to finally clear it up. 

Also, how I do I set up a distance timelapse for other than 10m? Is the variable a multiplier to 10, so 0.5 x10 = 5m or 5 x 10 =50m?

[EDIT]: I've did a run today on my bike to test it out. It appears to shoot per distance, but then when I stopped and enjoyed the scenery, it still took a picture as if it's a normal timelapse. I've set the distance to 0.5 with the hope it will give me 5m intervals.

**Reply by dnewman-gpsw** [reply]
Sorry the documentation is too thin for DIST.

For a single preset to use DIST, this must be its name, "Distance" will not be detected. Just setting $DIST=1 will globally enable the feature.  The command you want is ```mT$PRES="1,DIST"mTr5Xp4```

There are a lot in you command that is wrong, the camera doesn't have a 4K 8:7 timelapse mode (r4X), only 5.3K 8:7 (r5X), and setting a shutter angle for timelapse is not supported.

For a five meter interval use $DIST=5  (this is the part of the documentation that is missing), I will see that the docs are expanded.

You can't easily set a preset to 5 meter, but there is a work around using macros.

```*DSTA="$DIST=5'5m setting'"*DSTa="$DIST=0"mT$PRES="1,DSTA"mTr5Xp4```

DSTA  is your named preset, it runs ```$DIST=5'5m setting'```
DSTa  is the macro the is run when you leave the preset, so it will disable DIST.

This way you have have multiple presets at different distances

e.g.
```*DSTB="$DIST=25'25m setting'"*DSTb="$DIST=0"mT$PRES="1,DSTB"mTr5Xp4```

**Reply by bergie5737** [reply]
Thank you so much for the help. I've created the QR codes and scanned it. I will hopefully get a chance to test it today. Of course I don't know anything about macros. On a side note, in video mode I am able to pick 8:7 4k or 5.3k. That is why I selected that for my timelapse in the QR generator, but I can see from trying to edit a preset in timelapse, it is not available. So much I don't understand. So there is no way  I can keep the horizon locked in any timelapse mode?

**Reply by dnewman-gpsw** [reply]
All timelapse modes are unstabilized, so no horizon levelling.  Time warp is stabilized, not sure if that has horizon levelling.

**Reply by bergie5737** [reply]
Thanks. Got it! Coming back to the macro. I have now two presets, DSTA in slot 1 and DSTB in slot 2. Do I have to set this "mT$PRES="1,DSTB"mTr5Xp4" to mT$PRES="2,DSTB"mTr5Xp4" i.e. load the preset in slot 2? I base it on what ChatPT says: (The first two steps seems clear to me as well as step 4)

3: mT$PRES="1,DSTB":

    Puts the camera in Timelapse Video mode (mT).
    Temporarily saves the current settings to preset slot 1, naming it "DSTB". This means slot 1 will load and use the macro DSTB.

**Reply by dnewman-gpsw** [reply]
The GPT is making things up, as I still need to teach it about presets.  The 1 and 2 are icon numbers.

**Reply by dnewman-gpsw** [reply]
mT$PRES="1,DSTB"

mT - mode Timelapse 
$PRES="1,DSTB"  create a preset using icon 1 with name DSTB.

**Reply by bergie5737** [reply]
Once again thank you for clarifying. I went out cycling and it is working as expected now. Only thought, when I stop the distance lap and switch to a video, the overlay say "25m setting" I suppose that is normal. Also to get normal timelapses, do I have to power cycle or am I just doing something wrong? No too much of an issue as it does what I want!

**Reply by dnewman-gpsw** [reply]
It should only say "25m setting" when switch to the DSTB preset, not when switching to video. It should only display for a second. To get normal timelapse, select the Time Lapse Video present, that will be normal unless you scanned *DIST=1, which is a permanent setting (undo with *DIST=0.)
